services/web/server/src/simcore_service_webserver/conversations/_conversation_message_service.py

```diff
@@ -11,35 +11,65 @@
     ConversationMessagePatchDB,
     ConversationMessageType,
 )
+from models_library.projects import ProjectID
 from models_library.rest_ordering import OrderBy, OrderDirection
 from models_library.rest_pagination import PageTotalCount
 from models_library.users import UserID
 
+from ..projects._groups_repository import list_project_groups
+from ..users._users_service import get_users_in_group
+
+# Import or define SocketMessageDict
 from ..users.api import get_user_primary_group_id
 from . import _conversation_message_repository
+from ._socketio import (
+    notify_conversation_message_created,
+    notify_conversation_message_deleted,
+    notify_conversation_message_updated,
+)
 
 _logger = logging.getLogger(__name__)
 
 
+async def _get_recipients(app, project_id) -> set[UserID]:
+    groups = await list_project_groups(app, project_id=project_id)
```

Comment:
I think there are also socketio channels for groups. not sure you need to go all the way to the users.

Reply:
Yes, the issue is that a user might belong to multiple groups... in that case he/she will receive the notification multiple times.